### ISSUE TITLE
discussion(tech): Propose using Deno for the server

### ISSUE THREAD

**Issue by twilson63** [twilson63/permaweb-relay#2]
I would like to propose we use Deno/Javascript for the server technology.

## Why Deno?

* Security on the server
* Web Standards
* Built in Test library
* Built in Lint library
* Built in bundler

**Comment by VinceJuliano** [comment]
For sure a good idea, interacting with Arweave is smoothest through Javascript as well, can take a look into existing relays as well such as https://github.com/Cameri/nostream

**Comment by DanMacDonald** [comment]
Would love to do more deno stuff. It seems like the Arweave ecosystem packages are more ready for deno than they were a year ago.

**Comment by twilson63** [comment]
Yep, many of them support esm and are bundled for the browser, so arweave-js, webBundlr, and Warp should all work well for deno.

**Comment by twilson63** [comment]
Also, it is nice to have built in linters and testing tools, and the security features, for servers like this, so we know third party code can not do strange things to the underlying environment.

**Comment by twilson63** [comment]
Last point, we can package deno into an executable, so it makes it easier to distribute to bare-metal servers. Zero dependencies.

**Comment by VinceJuliano** [comment]
Yep the ability to build a single executable is vital I think, avoids installation of node etc... One thing I ran into that was an issue on previous Rust projects was cross compiling for example compiling from my windows machine and running on linux, will test out Deno and see if I can get an executable easily from one platform to another. Can always just compile on the needed platform if necessary.
